driver/tests.py

```diff
@@ -59,6 +61,42 @@ def test_portfolios():
         run_driver(parameters)
 
 
+def _get_portfolio_configs(portfolio: Path):
+    content = portfolio.read_text()
+    attributes = {}
+    try:
+        exec(content, attributes)
+    except Exception:
+        traceback.print_exc()
+        raise SyntaxError(
+            f"The portfolio {portfolio} could not be loaded.")
+    if "CONFIGS" not in attributes:
+        raise ValueError("portfolios must define CONFIGS")
+    return [config for _, config in attributes["CONFIGS"]]
+
+
+def _convert_to_standalone_config(config):
+    replacements = [
+        ("H_COST_TRANSFORM", "adapt_costs(one)"),
+        ("S_COST_TYPE", "one"),
+        ("BOUND", "infinity"),
+    ]
+    for index, part in enumerate(config):
+        for before, after in replacements:
+            part = part.replace(before, after)
+        config[index] = part
+    return config
+
+
+def test_portfolio_configs():
+    all_configs = set()
+    for portfolio in PORTFOLIOS.values():
+        configs = _get_portfolio_configs(Path(portfolio))
+        all_configs |= set(tuple(_convert_to_standalone_config(config)) for config in configs)
+    for config in set(all_configs):
```

Comment:
Do you need `set(...)` here?

Reply:
Apparently yes.

Reply:
Why? Isn't `all_configs` a set already? It is not modified in the loop, so I don't see why you'd need a copy.

Reply:
Ah, I thought you mean a different line. Thanks, I removed it.